mitodl/mit-open
# Add location field, populate for Sloan courses

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5129

### Description (What does it do?)
Adds a `location` field to `LearningResource` and `LearningResourceRun` and populates it for Sloan courses (locations for Professional Education courses will be handled after this one is merged in a [separate PR](https://github.com/mitodl/mit-open/pull/1210)).  Location isn't relevant [for other sources](https://docs.google.com/spreadsheets/d/1ZWeTcNgdhwd0E1s9ciaDTvouaRt5JVKFBY7UyoJ5GzY/edit?gid=0#gid=0).

The value for `LearningResource` will be based on the value of the next/best `LearningResourceRun`.


### How can this be tested?
-  Copy `SEE_` env settings from RC
- Run `./manage.py backpopulate_sloan_data`
- Run the following, you should get similar results:
  ```python
  from learning_resources.models import *
  for location in LearningResource.objects.values_list("location", flat=True).distinct():
      print(location)

  'Cambridge, MA', 
  'Melbourne, Australia',
  'Lausanne, Switzerland',
  ...etc
  ```
- The search page should still work, before and after running `recreate_index --all`

### Additional Context
